docs/reference/extensions/gnome-extension.rst

```diff
@@ -0,0 +1,253 @@
+
+.. collapse:: Environment variables
+
+    .. code-block:: yaml
+
+        environment:
+          SNAP_DESKTOP_RUNTIME: $SNAP/gnome-platform
+          GTK_USE_PORTAL: '1'
+
+
+Included layouts
+----------------
+
+This extension uses `layouts <https://snapcraft.io/docs/snap-layouts>`_ to access files
+on the host. The platform snap's gjs, webkit2gtk-4.0, and iso-codes are used so they
+don't need to be packaged as part of the snap and would greatly inflate the size.
+
+.. tabs::
+
+    .. group-tab:: core24
+
+        .. collapse:: Included layouts
+
+            .. code-block:: yaml
+
+                layout:
+                  /usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0:
+                    bind: $SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0
+                  /usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1:
+                    bind: $SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1
+                  /usr/share/xml/iso-codes:
+                    bind: $SNAP/gnome-platform/usr/share/xml/iso-codes
+                  /usr/share/libdrm:
+                    bind: $SNAP/gpu-2404/libdrm
+                  /usr/share/drirc.d:
+                    symlink: $SNAP/gpu-2404/drirc.d
+                  /usr/share/X11/XErrorDB:
+                    symlink: $SNAP/gpu-2404/X11/XErrorDB
+
+    .. group-tab:: core22
+
+        .. collapse:: Included layouts
+
+            .. code-block:: yaml
+
+                layout:
+                  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libgweather-4:
+                    symlink: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libgweather-4
+                  /usr/lib/evolution-data-server:
+                    symlink: $SNAP/usr/lib/evolution-data-server
+                  /usr/bin/gnome-control-center:
+                    symlink: $SNAP/usr/bin/gnome-control-center
+                  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0:
+                    bind: $SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0
+                  /usr/share/xml/iso-codes:
+                    bind: $SNAP/gnome-platform/usr/share/xml/iso-codes
+                  /usr/share/libdrm:
+                    bind: $SNAP/gnome-platform/usr/share/libdrm
+
+
+Example expanded project file
+-----------------------------
+
+Here's an example of the result of Snapcraft expanding a core24-based project file, as
+immediately prior to build. It demonstrates the added plugs, packages, variables, and
+layouts, that the GNOME extension includes in a project.
+
+The original file was the `GNOME System Monitor snap
+<https://snapcraft.io/gnome-system-monitor>`_. This text contains the difference between
+the original file and the output of the :ref:`snapcraft expand-extensions
+<ref_commands_expand-extensions>` command. Some of the text has been altered for ease of
+reading.
+
+.. collapse:: Expanded project file for GNOME System Monitor
```

Comment:
@lengau Any low-hanging core22 projects I could include?

Reply:
I was going to recommend gnome 2048, but the gnome devs do a good job updating to the latest bases.

You could grab the copy of this snapcraft.yaml from last year:
https://github.com/ubuntu/gnome-2048/commit/38e48d9bab9b29c27e48bc2346b78ede323d5dc2,

Reply:
I added the core22 version of System Monitor: [44662f7](https://github.com/canonical/snapcraft/pull/5212/commits/44662f7cd24bc2b2cf376fb3e7d9de5a43d46f4f)